src/common/transformations/src/transformations/symbolic_transformations/symbol_optimization.cpp

```diff
@@ -354,12 +355,99 @@ void save_shape_sources(const std::shared_ptr<ov::Node>& op, STS_map& symbol_sha
         }
     }
 }
+
+struct OutputValue {
+    std::vector<ov::Any> value;
+
+    bool operator==(const OutputValue& other) const {
+        if (value.size() != other.value.size())
+            return false;
+        for (size_t i = 0; i < value.size(); ++i)
+            if (value.at(i) != other.value.at(i))
+                return false;
+        return true;
+    }
+
+    bool operator<(const OutputValue& other) const {
+        return std::lexicographical_compare(
+            std::begin(value),
+            std::end(value),
+            std::begin(other.value),
+            std::end(other.value),
+            [](const ov::Any& a, const ov::Any& b) {
+                // each element is either a symbol or an integer. in case they differ any integer is less than a symbol.
+                if (a.is<std::shared_ptr<ov::Symbol>>() && b.is<std::shared_ptr<ov::Symbol>>())
+                    return a.as<std::shared_ptr<ov::Symbol>>() < b.as<std::shared_ptr<ov::Symbol>>();
+                if (a.is<int64_t>() && b.is<int64_t>())
+                    return a.as<int64_t>() < b.as<int64_t>();
+                return a.is<int64_t>();
+            });
+    }
+
+    static std::pair<bool, OutputValue> make(const ov::Output<ov::Node>& output) {
+        auto symbols = output.get_tensor().get_value_symbol();
+        if (symbols.empty() || symbols.size() == 1)
+            return {false, OutputValue()};
+        if (std::all_of(symbols.begin(), symbols.end(), [](const std::shared_ptr<ov::Symbol>& s) {
+                return s != nullptr;
+            })) {
+            std::vector<ov::Any> symbols_as_any(symbols.size(), nullptr);
+            for (size_t i = 0; i < symbols.size(); ++i)
+                symbols_as_any[i] = ov::symbol::ancestor_of(symbols[i]);
+            return {true, {symbols_as_any}};
+        } else {
+            const auto& lower = output.get_tensor().get_lower_value();
+            const auto& upper = output.get_tensor().get_upper_value();
+            const auto& et = output.get_element_type();
```

Comment:
why is there a restriction on element types?
do we need this check in 1st "if" branch?

Reply:
it is here to only work with dimention-like element types. no, there is no need to restrict it in other if branch, since there we only work with symbols.